RecoLocalMuon/RPCRecHit/src/RPCPointProducer.cc

```diff
@@ -25,44 +25,27 @@
 
 // user include files
 
-RPCPointProducer::RPCPointProducer(const edm::ParameterSet& iConfig)
+RPCPointProducer::RPCPointProducer(const edm::ParameterSet& iConfig) :
+  cscSegments(  consumes<CSCSegmentCollection>(iConfig.getParameter<edm::InputTag>("cscSegments")) ),
+  dt4DSegments( consumes<DTRecSegment4DCollection>(iConfig.getParameter<edm::InputTag>("dt4DSegments")) ),
+  tracks(       consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks")) ),
+  tracks_(      iConfig.getParameter<edm::InputTag>("tracks") ),
+  incldt(       iConfig.getUntrackedParameter<bool>("incldt", true) ),
+  inclcsc(      iConfig.getUntrackedParameter<bool>("inclcsc", true) ),
+  incltrack(    iConfig.getUntrackedParameter<bool>("incltrack", true) ), 
+  debug(        iConfig.getUntrackedParameter<bool>("debug", false) ),
+  MinCosAng(    iConfig.getUntrackedParameter<double>("MinCosAng", 0.95) ),
+  MaxD(         iConfig.getUntrackedParameter<double>("MaxD", 80.) ),
+  MaxDrb4(      iConfig.getUntrackedParameter<double>("MaxDrb4", 150.) ),
+  ExtrapolatedRegion( iConfig.getUntrackedParameter<double>("ExtrapolatedRegion", 0.5) ),
+  trackTransformerParam( iConfig.getParameter<edm::ParameterSet>("TrackTransformer") )
 {
-  cscSegments = consumes<CSCSegmentCollection>(iConfig.getParameter<edm::InputTag>("cscSegments"));
-  
-  dt4DSegments = consumes<DTRecSegment4DCollection>(iConfig.getParameter<edm::InputTag>("dt4DSegments"));
-  tracks = consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks"));
-  tracks_ = iConfig.getParameter<edm::InputTag>("tracks");
-
-  debug=iConfig.getUntrackedParameter<bool>("debug",false);
-  incldt=iConfig.getUntrackedParameter<bool>("incldt",true);
-  inclcsc=iConfig.getUntrackedParameter<bool>("inclcsc",true);
-  incltrack=iConfig.getUntrackedParameter<bool>("incltrack",true);
-  MinCosAng=iConfig.getUntrackedParameter<double>("MinCosAng",0.95);
-  MaxD=iConfig.getUntrackedParameter<double>("MaxD",80.);
-  MaxDrb4=iConfig.getUntrackedParameter<double>("MaxDrb4",150.);
-  ExtrapolatedRegion=iConfig.getUntrackedParameter<double>("ExtrapolatedRegion",0.5);
-
   produces<RPCRecHitCollection>("RPCDTExtrapolatedPoints");
```

Comment:
based on what's below, the produces methods should be hidden behind ``` if (incldt) ``` etc.
Otherwise, branches are declared but are never filled.
since it's been running, I guess the flags are always set